engines/lilliput/script.cpp

```diff
@@ -0,0 +1,3390 @@
+/* 0x5c */	{ "OC_sub186A1", 3, kGetValue1, kImmediateValue, kImmediateValue, kNone, kNone },  //TODO
+/* 0x5d */	{ "OC_sub186E5_snd", 2, kGetValue1, kImmediateValue, kNone, kNone, kNone }, 
+/* 0x5e */	{ "OC_sub1870A_snd", 2, kgetPosFromScript, kImmediateValue, kNone, kNone, kNone }, 
+/* 0x5f */	{ "OC_sub18725_snd", 1, kGetValue1, kNone, kNone, kNone, kNone }, 
+/* 0x60 */	{ "OC_sub18733_snd", 1, kGetValue1, kNone, kNone, kNone, kNone }, 
+/* 0x61 */	{ "OC_sub1873F_snd", 1, kgetPosFromScript, kNone, kNone, kNone, kNone }, 
+/* 0x62 */	{ "OC_sub18746_snd", 1, kImmediateValue, kNone, kNone, kNone, kNone }, 
+/* 0x63 */	{ "OC_sub1875D_snd", 0, kNone, kNone, kNone, kNone, kNone }, 
+/* 0x64 */	{ "OC_setCharacterMapColor", 2, kGetValue1, kImmediateValue, kNone, kNone, kNone }, 
+/* 0x65 */	{ "OC_initGameAreaDisplay", 0, kNone, kNone, kNone, kNone, kNone }
+};
+
+Common::String LilliputScript::getArgumentString(KValueType type, ScriptStream& script) {
+	
+	Common::String str;
+	if (type == kImmediateValue) {
+		str =  Common::String::format("0x%x", script.readUint16LE());
+	} else if (type == kGetValue1) {
+		int val = script.readUint16LE();
+		if (val < 1000) { 
+			str = Common::String::format("0x%x", val);
+		} else if (val > 1004) { 
+			str = Common::String::format("getValue1(0x%x)", val);
+		} else if (val == 1000) {
+			str = Common::String("_selectedCharacterId");
+		} else if (val == 1001) {
+			str = Common::String("characterIndex");
+		} else if (val == 1002) {
+			str = Common::String("_word16F00_characterId");
+		} else if (val == 1003) {
+			str = Common::String("_currentCharacterVariables[6]");
+		} else if (val == 1004) {
+			str = Common::String("_word10804");
+		}
+	} else if (type == kgetPosFromScript) {
+		int curWord = script.readUint16LE();
+		int tmpVal = curWord >> 8;
+		switch(tmpVal) {
+	case 0xFF:
+		str = "(_rulesBuffer2_13[currentCharacter],_rulesBuffer2_14[currentCharacter])";
+		break;
+	case 0xFE: {
+		int index = curWord & 0xFF;
+		assert((index >= 0) && (index < 40));
+		str = Common::String::format("_vm->_rulesBuffer2_13[%d],_vm->_rulesBuffer2_14[%d]", index, index);
+		break;
+			   }
+	case 0xFD:
+		str = "_currentScriptCharacterPosition";
+		break;
+	case 0xFC: {
+		int index = curWord & 0xFF;
+		assert(index < 40);
+		str = Common::String::format("(characterPositionTileX[%d], characterPositionTileY[%d])", index, index);
+		break;
+			   }
+	case 0xFB: {
+		str = "(characterPositionTileX[_word16F00_characterId], characterPositionTileY[_word16F00_characterId])";
+		break;
+			   }
+	case 0xFA:
+		str = Common::String::format("(_array10999PosX[currentCharacter], _array109C1PosY[currentCharacter])");
+		break;
+	case 0xF9:
+		str = Common::String::format("(_currentCharacterVariables[4], _currentCharacterVariables[5])");
+		break;
+	case 0xF8: {
+		int index = curWord & 0xFF;
+		assert((index >= 0) && (index < 40));
+		str = Common::String::format("_vm->_rulesBuffer12Pos3[%d]", index);
```

Comment:
Looks like a `break` is missing here. If this fallthrough really is intentional it should be marked as such with `// fall through`.